packages/graphback-datasync/src/services/createDataSyncService.ts

```diff
@@ -0,0 +1,33 @@
+import { ModelDefinition } from '@graphback/core';
+import { PubSubEngine } from 'graphql-subscriptions';
+import { CRUDService, CRUDServiceConfig, GraphbackCRUDService } from '@graphback/runtime';
+import { isDataSyncModel } from '../util';
+import { DataSyncProvider } from '../providers';
+import { DataSyncCRUDService } from './DataSyncCRUDService';
+
+export interface CreateDataSyncCRUDServiceOptions {
+  /**
+   * PubSub implementation for creating subscriptions
+   */
+  pubSub: PubSubEngine
+}
+
+/**
+ *
+ * @param config
+ */
+export function createDataSyncCRUDService(config: CreateDataSyncCRUDServiceOptions): (...args: any[]) => GraphbackCRUDService {
+  return (model: ModelDefinition, dataProvider: DataSyncProvider): GraphbackCRUDService => {
+
+    const serviceConfig: CRUDServiceConfig = {
+      ...config,
+      crudOptions: model.crudOptions
+    }
+
+    if (isDataSyncModel(model)) {
+      return new DataSyncCRUDService(model.graphqlType.name, dataProvider, serviceConfig)
+    } else {
+      return new CRUDService(model.graphqlType.name, dataProvider, serviceConfig)
```

Comment:
DataSyncProvider CRUD methods differ quite a lot from standard MongoDBDataProvider, but there is no way to determine if this is Mongo/Postgres.

So we will still need to add serviceOverrides to the runtime API.

Reply:
cc @wtrocki @machi1990

Reply:
This is bad.. DataSync that should be evaluating on top of graphback is leaking to graphback core. This will never have happened if we would work on separate repository.

Reply:
I will work on this with @ssd71 to find best approach